### PR TITLE
validate applab data storage blocks.

### PR DESCRIPTION
**Pull request by davidsbailey**

### CODE REVIEW

File: apps/src/applab/applab.js [hunk 1]
```diff
@@ -2905,6 +2946,11 @@ Applab.handleReadRecords = function(successCallback, records) {
 };
 
 Applab.updateRecord = function (opts) {
+  apiValidateType(opts, 'updateRecord', 'table', opts.table, 'string');
+  apiValidateType(opts, 'updateRecord', 'record', opts.record, 'object');
+  apiValidateTypeAndRange(opts, 'updateRecord', 'record.id', opts.record.id, 'number', 1, Infinity);
```
Comment (cpirich) [inline apps/src/applab/applab.js]: Do you want to validate that the number is an integer as well? Perhaps a new "integer" type that behaves like "number" but a little different?

**Comment by cpirich** [comment]
Looks good. My other comment is that you should decide whether or not it is ok for other types to be coerced into strings. So perhaps table could be 1 instead of "1"? I'm thinking of changing some string validations to simply check that it is not undefined or object.

File: apps/src/applab/applab.js (hunk 1 above)
Comment (davidsbailey) [inline apps/src/applab/applab.js]: It seems unlikely that a student would end up with a float here. However, borrowing from your other suggestion, I would propose allowing numerical strings (e.g. '11') anywhere we require a number (see updates to this PR). Let me know what you think.

**Comment by davidsbailey** [comment]
Good point about allowing coerced types. In this particular example, it seems more likely that if a table is named 1 (as opposed to '1' or 'mytable') that it would be due to a programming error, so it seems ok to leave this as-is for now.

If you do change some string validation to accept any types besides 'undefined' or 'object', you may also want to reject type 'function'.

**Comment by cpirich** [comment]
LGTM

**Pull request merged**